assets/src/scripts/components/graph/view/construction.js

```diff
@@ -128,6 +131,55 @@ const drawPatterns = function (elem) {
         });
 };
 
+const drawWellBorehole =  function drawWellHole(store, container, opts, elements) {
+    // get the sate from the store
+    const state = store.getState();
+    // get the measured depth of the well
+    const rawDepth    = getSiteWellDepth(opts.agencyCode, opts.siteId)(state);
+    // scale the depth to the graph size
+    const scaleDepth  = getScaleY(opts, 'construction')(state)(rawDepth);
```

Comment:
A pattern in the code, which probably isn't obvious, is that the render functions don't know about Redux - they are passed in what they need to render, and the main `graph/view/index.js` entrypoint is where all the Redux lookups are made. This makes it easier to write tests for, eliminating the need to mock up Redux state.

If that's a possibility here, that would be nice.. it might make sense to export this function and call it from `index.js` rather than the default export, below.

Reply:
That's odd, I followed the pattern of the drawElement method. Or possibly more accurate, I needed to pass the store to drawElement and thus used it. Should both be refactored? Or am I missing the point? (I do not mind refactoring both if needed.)

Reply:
In drawElement, the store is passed in so an event can ("mouseenter") can dispatch an action to the store rather than getting state.

Reply:
sure. but I need the store in my method if I am to pass it to drawElement.

Reply:
True enough but you also retrieve state information in drawWellBoreHole. I haven't looked at this in detail so I don't know what would be needed to refactor this. I just wanted to point out the differences between how the store is used in drawElement vs drawWellBoreHole.

Reply:
Right. and I was trying to say that the reason I passed in the store instead of the state was because drawElement needed it. I wrote the drawBorehole while the drawElement was already there.

Reply:
Mary, Dan screen shared via slack to clear this up. I have a much better idea what could be done.